scala/src/main/org/apache/spark/streaming/api/csharp/CSharpDStream.scala

```diff
@@ -343,3 +347,149 @@ class CSharpConstantInputDStream(ssc_ : StreamingContext, rdd: RDD[Array[Byte]])
 
   val asJavaDStream: JavaDStream[Array[Byte]] = JavaDStream.fromDStream(this)
 }
+
+case class RddPreComputeRecord[T] (
+    rddSeqNum: Long,
+    userInfo: Any,
+    rdd: RDD[T])
+
+/*
+ * Slurp RDDs, apply cache and pre-compute operation asynchronously, then spit them out
+ */
+class RddPreComputeCache[T](
```

Comment:
Consider renaming. Cache implementations typically do not operate on the data. That is not the case with this implementation. Suggestion - RddPreComputeProcessor